Build failure: goldendict-ng

### Steps To Reproduce

Build *goldendict-ng* (24.09.0) on current nixos-unstable 

### Build log

<details>
<summary>Part of build log (log where is the error)</summary>

```
[ 76%] Building CXX object CMakeFiles/goldendict.dir/src/resourceschemehandler.cc.o
[ 77%] Building CXX object CMakeFiles/goldendict.dir/src/speechclient.cc.o
[ 77%] Building CXX object CMakeFiles/goldendict.dir/src/splitfile.cc.o
[ 78%] Building CXX object CMakeFiles/goldendict.dir/src/termination.cc.o
[ 78%] Building CXX object CMakeFiles/goldendict.dir/src/texttospeechsource.cc.o
[ 79%] Building CXX object CMakeFiles/goldendict.dir/src/tiff.cc.o
/build/source/src/multimediaaudioplayer.cc: In member function 'void MultimediaAudioPlayer::audioOutputChange()':
/build/source/src/multimediaaudioplayer.cc:40:9: error: invalid use of incomplete type 'class QDebug'
   40 |   qDebug() << "audio device changed";
      |         ^
In file included from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qglobal.h:47,
                 from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qnamespace.h:12,
                 from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qbytearray.h:9,
                 from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/QByteArray:1,
                 from /build/source/src/multimediaaudioplayer.cc:6:
/nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qtypeinfo.h:15:7: note: forward declaration of 'class QDebug'
   15 | class QDebug;
      |       ^~~~~~
make[2]: *** [CMakeFiles/goldendict.dir/build.make:1714: CMakeFiles/goldendict.dir/src/multimediaaudioplayer.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/build/source/src/main.cc: In function 'int main(int, char**)':
/build/source/src/main.cc:566:72: warning: 'static QString QLibraryInfo::location(LibraryLocation)' is deprecated: Use path() [-Wdeprecated-declarations]
  566 |     qtTranslator.load( "qt_extra_" + localeName, QLibraryInfo::location( QLibraryInfo::TranslationsPath ) );
      |                                                  ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/QtCore:146,
                 from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtNetwork/QtNetworkDepends:3,
                 from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtNetwork/QtNetwork:3,
                 from /build/source/src/article_netmgr.hh:7,
                 from /build/source/src/ui/mainwindow.hh:17,
                 from /build/source/src/main.cc:6:
/nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qlibraryinfo.h:49:20: note: declared here
   49 |     static QString location(LibraryLocation location)
      |                    ^~~~~~~~
/build/source/src/main.cc:566:22: warning: ignoring return value of 'bool QTranslator::load(const QString&, const QString&, const QString&, const QString&)', declared with attribute 'nodiscard' [-Wunused-result]
  566 |     qtTranslator.load( "qt_extra_" + localeName, QLibraryInfo::location( QLibraryInfo::TranslationsPath ) );
      |     ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/QtCore:288:
/nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qtranslator.h:33:24: note: declared here
   33 |     [[nodiscard]] bool load(const QString & filename,
      |                        ^~~~
/build/source/src/main.cc:570:18: warning: ignoring return value of 'bool QTranslator::load(const QString&, const QString&, const QString&, const QString&)', declared with attribute 'nodiscard' [-Wunused-result]
  570 |   translator.load( Config::getLocDir() + "/" + localeName );
      |   ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/bgfalfi93kbn8j1wfwz0x0dnk1wx9wdp-qtbase-6.8.0/include/QtCore/qtranslator.h:33:24: note: declared here
   33 |     [[nodiscard]] bool load(const QString & filename,
      |                        ^~~~
make[1]: *** [CMakeFiles/Makefile2:94: CMakeFiles/goldendict.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

</details>

full log: https://gist.github.com/Blezz-tech/2d363d0d81e92219cd0e4c749b041b04

### Additional context

fix build failure in upstream

> fix: Qt6.8 compiling error -> QDebug

https://github.com/xiaoyifang/goldendict-ng/releases/tag/v24.11.0-alpha.b19b3191

I used last daily release and it build successfully

```nix
(self: super: {
  goldendict-ng = super.goldendict-ng.overrideAttrs (oldAttrs: {
    src = super.fetchFromGitHub {
      owner = "xiaoyifang";
      repo = "goldendict-ng";
      rev = "v24.11.0-alpha.870ff080";
      hash = "sha256-/myO1tDdE3/yiMSmGiGfKqBhZekIsGheJ/VExyhQ+mg=";
    };
  });
})
```

### Notify maintainers

@slbtty
@michojel

### Metadata

```
~/downloads took 3s ❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.58, NixOS, 24.11 (Vicuna), 24.11.20241025.18536bf`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/wb6agba4kfsxpbnb5hzlq58vkjzvbsk6-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/goldendict-ng/default.nix
+++ b/pkgs/applications/misc/goldendict-ng/default.nix
@@ -30,13 +30,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "goldendict-ng";
-  version = "24.09.0";
+  version = "24.09.1";
 
   src = fetchFromGitHub {
     owner = "xiaoyifang";
     repo = "goldendict-ng";
-    rev = "v${finalAttrs.version}-Release.316ec900";
-    hash = "sha256-LriKJLjqEuD0v8yjoE35O+V6oUX2jhWGFguqlXaDlQA=";
+    rev = "v${finalAttrs.version}-Release.ca9dd133";
+    hash = "sha256-HvXC9fNLDZAEtXNjzFmaKz9Ih3n4Au69NPMuVeCiQPk=";
   };
 
   nativeBuildInputs = [ pkg-config cmake wrapQtAppsHook wrapGAppsHook3 ];

